datafusion/src/scalar.rs

```diff
@@ -65,8 +65,9 @@ pub enum ScalarValue {
     Binary(Option<Vec<u8>>),
     /// large binary
     LargeBinary(Option<Vec<u8>>),
-    /// list of nested ScalarValue
-    List(Option<Vec<ScalarValue>>, DataType),
+    /// list of nested ScalarValue (boxed to reduce size_of(ScalarValue))
+    #[allow(clippy::box_vec)]
+    List(Option<Box<Vec<ScalarValue>>>, Box<DataType>),
```

Comment:
Here is the change -- the rest of the PR is just follow on work from this

Reply:
This is an interesting optimization 👍

Reply:
`Box<[ScalarValue]>` via `Vec::into_boxed_slice` would also be an option and would remove one pointer indirection, with the downside that data would need to be copied if the vec has excess capacity. @alamb do you think this would be worth exploring? I could prepare a PR since I already started looking to the usages.

Reply:
@jhorstmann  I think using `Vec::into_boxed_slice` would be just fine. I don't think `ScalarValues` are often (ever?) updated after creation so using a boxed slice seems like a good idea